GW2EIEvtcParser/EncounterLogic/Raids/W7/PeerlessQadim.cs

```diff
@@ -286,24 +317,46 @@ internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog lo
                     }
                     break;
                 case (int)ArcDPSEnums.TrashID.EntropicDistortion:
-                    //sapping surge, red tether
+                    // Sapping Surge, bad red tether
                     AddTetherDecorations(log, target, replay, SappingSurge, "rgba(255, 0, 0, 0.4)");
+
+                    // Stun icon
+                    IEnumerable<Segment> stun = target.GetBuffStatus(log, Stun, log.FightData.FightStart, log.FightData.FightEnd).Where(x => x.Value > 0);
+                    foreach (Segment s in stun)
+                    {
+                        replay.AddOverheadIcon(s, target, BuffImages.Stun);
+                    }
+
+                    // Spawn animation
                     Point3D firstEntropicPosition = replay.PolledPositions.FirstOrDefault();
+                    int radiusAnomaly = (int)target.HitboxWidth / 2;
                     if (firstEntropicPosition != null)
                     {
-                        replay.AddDecorationWithGrowing(new CircleDecoration(300, (start - 5000, start), "rgba(255, 0, 0, 0.3)", new PositionConnector(firstEntropicPosition)), start);
+                        replay.AddDecorationWithGrowing(new CircleDecoration(radiusAnomaly, (start - 5000, start), "rgba(255, 0, 0, 0.3)", new PositionConnector(firstEntropicPosition)), start);
                     }
                     break;
                 case (int)ArcDPSEnums.TrashID.BigKillerTornado:
                     replay.Decorations.Add(new CircleDecoration(450, (start, end), "rgba(255, 150, 0, 0.4)", new AgentConnector(target)));
                     break;
                 case (int)ArcDPSEnums.TrashID.FriendlyPeerlessQadimPylon:
+                    // Red tether from Qadim to the Pylon during breakbar
+                    List<AbstractBuffEvent> breakbarBuffs = GetFilteredList(log.CombatData, QadimThePeerlessBreakbarTargetBuff, target, true, true);
+                    replay.AddTether(breakbarBuffs, "rgba(255, 0, 0, 0.4)");
                     break;
                 case (int)ArcDPSEnums.TrashID.HostilePeerlessQadimPylon:
                     break;
                 case (int)ArcDPSEnums.TrashID.EnergyOrb:
                     replay.Decorations.Add(new CircleDecoration(200, (start, end), "rgba(0, 255, 0, 0.3)", new AgentConnector(target)));
                     break;
+                case (int)ArcDPSEnums.TrashID.GiantQadimThePeerless:
+                    // Trim the first giant Qadim, it exists since log start.
+                    AgentItem firstGiantQadim = log.AgentData.GetNPCsByID(ArcDPSEnums.TrashID.GiantQadimThePeerless).OrderBy(x => x.FirstAware).FirstOrDefault();
```

Comment:
How many GiantQadimThePeerless are there? Why each one is trimmed based on the first?
If there are multiple ones with only one being valid, it would be better to identify it in EIEvtcParse and override its species with a custom one.

Reply:
There are 3, first one is there since the encounter start and this trims its visibility only during the 40% phase, they disappear after they get hit by the sak, following ones appear again when 30% and 20% happen and disappear when completing the mechanic